Sources/MapboxNavigation/CameraController.swift

```diff
@@ -0,0 +1,302 @@
+import MapboxDirections
+import MapboxCoreNavigation
+import MapboxMaps
+
+
+class CameraController: NavigationComponent, ViewCycleObserver {
```

Comment:
IMO current PR should be rebased onto `maxim/camera-viewport-api` so that we start using `NavigationCamera` here as well. At this point `CameraController` uses properties, which are no longer present in mentioned PR (e.g. `tiltedCamera`, `contentInsets` are now prepared inside `NavigationViewportDataSource`, transitions are executed using `CameraAnimator` instead of `setCamera` calls).

`maxim/camera-viewport-api` also modifies `RouteMapViewController` a bit changes will be needed again to accomodate those viewport camera changes. @Udumft, let me know if you need more clarifications there.

Reply:
That makes sense. I'll try rebasing and see how it goes